### PR TITLE
Update public types

### PR DESCRIPTION
**Pull request by sisuresh**
Required for https://github.com/stellar/rs-soroban-env/issues/451. It updates public_types to match soroban-auth, with the main change being the added Invoker.

This may also resolve https://github.com/stellar/rs-soroban-env/issues/472 for now, but the auth code in the token contract is still mostly outdated (!!!) Issue for this - https://github.com/stellar/rs-soroban-env/issues/502. It's just the old auth code with some changes made to include the invoker. I was able to test auth using #686 (contract interface spec) and it worked, but this needs to be tested more thoroughly. 

Once this is merged, @dmkozh can test this with the tests he already has written and merge them in.

### CODE REVIEW

File: soroban-env-host/src/native_contract/token/admin.rs [hunk 1]
```diff
@@ -20,29 +18,12 @@ pub fn write_administrator(e: &Host, id: Identifier) -> Result<(), Error> {
...
-        (Identifier::Account(admin_id), Signature::Account(signatures)) => {
-            if admin_id.compare(&signatures.account_id)? != Ordering::Equal {
-                Err(Error::ContractError)
-            } else {
-                Ok(())
-            }
-        }
-        _ => Err(Error::ContractError),
+
+    let id = Signature::get_identifier(auth, e)?;
```
Comment (leighmcculloch) [inline soroban-env-host/src/native_contract/token/admin.rs:22]: Is there a reason you can't do this here?
```suggestion
    let id = auth.get_identifier(e)?;
```
Just curious.

File: soroban-env-host/src/native_contract/token/admin.rs [hunk 2]
```diff
@@ -20,29 +18,12 @@ pub fn write_administrator(e: &Host, id: Identifier) -> Result<(), Error> {
...
+    //TODO:IS THIS SAFE???
+    if id != admin {
```
Comment (leighmcculloch) [inline soroban-env-host/src/native_contract/token/admin.rs:24]: The values should be comparable like all Soroban values. If it wasn't safe here, it wouldn't be safe elsewhere. Can you expand on the concern?

**Review by leighmcculloch — APPROVED** [review]

File: soroban-env-host/src/native_contract/token/admin.rs (hunk 2 above)
Comment (sisuresh) [reply soroban-env-host/src/native_contract/token/admin.rs:24]: Argh I removed this comment but forgot to force push my local changes before marking this ready for review.

File: soroban-env-host/src/host.rs [hunk 3]
```diff
@@ -901,8 +901,9 @@ impl Host {
         })?;
 
         let lk = LedgerKey::Account(LedgerKeyAccount {
-            account_id: AccountId(PublicKey::PublicKeyTypeEd25519(self.to_u256(account_id)?)),
+            account_id: self.to_account_id(account)?,
```
Comment (dmkozh) [inline soroban-env-host/src/host.rs:904]: Please make `account_id` building consistent with `to_trustline_key`. Also I'm not sure if this one is correct, we want `AccountId` xdr here, don't we?

Comment (sisuresh) [reply soroban-env-host/src/host.rs:904]: Just to be clear, are you saying I should add a `to_account_key` method? And `to_account_id` returns the xdr `AccountId`. It's a little confusing, but the `AccountId` in base_types is only used in the token contract.

Comment (dmkozh) [reply soroban-env-host/src/host.rs:904]: If `to_account_id` returns the xdr, then why don't we use it in `to_trustline_key`? We build account ids for both trustline user and issuer manually there.

Comment (dmkozh) [reply soroban-env-host/src/host.rs:904]: Do we maybe have different inputs to these methods? Maybe we should document what are the expected `Object` types, because the code is pretty hard to follow currently (or maybe use explicit types instead of `Object`s where possible).

Comment (sisuresh) [reply soroban-env-host/src/host.rs:904]: Yeah that's a good point. I added `to_account_id` after `to_u256_from_account` and missed this. My goal here was to have all of these methods accept `HostObject::AccountId` instead of `HostObject::Bytes`, which is what `to_u256` accepts. I updated `to_trustline_key` to use `account_id`.

I agree we should add documentation about the expected `Object` types. I'll open an issue for this.

Comment (dmkozh) [reply soroban-env-host/src/host.rs:904]: Thanks! The change LGTM now.
